src/test/java/am/ik/yavi/core/ConstraintOnClassTest.java

```diff
@@ -0,0 +1,71 @@
+package am.ik.yavi.core;
+
+import am.ik.yavi.User;
+import am.ik.yavi.builder.ValidatorBuilder;
+import am.ik.yavi.constraint.CharSequenceConstraint;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import static com.google.common.truth.Truth.assertThat;
+
+public class ConstraintOnClassTest {
+
+	@ParameterizedTest
+	@MethodSource("provideValidators")
+	void testConstraintOnConditionClass(Validator<User> validator) {
+		User validAdmin = new Admin("admin123", "admin@gmail", 27, "yavi123");
+		User invalidAdmin = new Admin("Niraz", "niraz@gmail", 23, "user");
+
+		assertThat(validator.validate(validAdmin).isValid()).isTrue();
+		assertThat(validator.validate(invalidAdmin).isValid()).isFalse();
+	}
+
+	@ParameterizedTest
+	@MethodSource("provideValidators")
+	void testConstraintOnNonConditionClass(Validator<User> validator) {
+		User validUser = new User("Rawad", "rawad@gmail", 25);
+		User invalidUser = new User("Almog", "almog@gmail", 19);
+
+		assertThat(validator.validate(validUser).isValid()).isTrue();
```

Comment:
It is not enough to simply verify the results of validation. Please also verify the contents of the constraints.